### PR TITLE
Make pluginconfig optional in checkconfig.

### PR DESCRIPTION
**Pull request by amwat**
For folks that don't use any plugins.

/cc @krzyzacy

### CODE REVIEW

File: prow/cmd/checkconfig/main.go [hunk 1]
```diff
@@ -176,7 +176,7 @@ func main() {
 			errs = append(errs, err)
 		}
 	}
-	if o.warningEnabled(validateOwnersWarning) {
+	if pcfg!=nil && o.warningEnabled(validateOwnersWarning) {
```
Comment (krzyzacy) [inline prow/cmd/checkconfig/main.go]: space nit?

File: prow/cmd/checkconfig/main.go [hunk 2]
```diff
@@ -146,10 +143,13 @@ func main() {
 	cfg := configAgent.Config()
 
 	pluginAgent := plugins.ConfigAgent{}
+	var pcfg *plugins.Configuration
 	if err := pluginAgent.Load(o.pluginConfig); err != nil {
```
Comment (krzyzacy) [inline prow/cmd/checkconfig/main.go]: if `o.pluginConfig` is empty, then `pcfg = nil`, we still want to catch when it fails to load a non-empty plugins config

**Comment by krzyzacy** [comment]
cc @cjwagner @stevekuznetsov - not all deployments have plugins :-)

File: prow/cmd/checkconfig/main.go (hunk 1 above)
Comment (amwat) [reply prow/cmd/checkconfig/main.go]: done

File: prow/cmd/checkconfig/main.go (hunk 2 above)
Comment (amwat) [reply prow/cmd/checkconfig/main.go]: done

Comment (krzyzacy) [reply prow/cmd/checkconfig/main.go]: hummm, I think here should be something like
```
if o.pluginConfig != "" {
  if err := pluginAgent.Load(o.pluginConfig); err != nil {
    Fatal
  }
  pcfg = pluginAgent.Config()
}
```

Comment (amwat) [reply prow/cmd/checkconfig/main.go]: done

File: prow/cmd/checkconfig/main.go [hunk 3]
```diff
@@ -295,16 +295,18 @@ func validateTideRequirements(cfg *config.Config, pcfg *plugins.Configuration) e
...
-		err := ensureValidConfiguration(
-			pluginConfig.plugin,
-			pluginConfig.label,
-			pluginConfig.matcher.verb,
-			pluginConfig.config,
-			overallTideConfig,
-			enabledOrgReposForPlugin(pcfg, pluginConfig.plugin, pluginConfig.external),
-		)
-		validationErrs = append(validationErrs, err)
+	if pcfg != nil {
```
Comment (stevekuznetsov) [inline prow/cmd/checkconfig/main.go]: This seems wrong - why bother calling to validate tide requirements if we don't execute anything?

Comment (stevekuznetsov) [reply prow/cmd/checkconfig/main.go]: @amwat

Comment (krzyzacy) [reply prow/cmd/checkconfig/main.go]: lol yeah... you can just skip this entire `validateTideRequirements` check

Comment (amwat) [reply prow/cmd/checkconfig/main.go]: cleaned up in https://github.com/kubernetes/test-infra/pull/11416 :)

**Pull request merged**